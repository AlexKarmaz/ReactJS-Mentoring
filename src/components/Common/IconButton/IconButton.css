.iconButton {
    padding: 0;
    margin: 0;
    cursor: pointer;
    outline: none;
    justify-content: center;
    display: flex;
    align-items: center;
    border-radius: 50%;
    border: #232323;
    background-color: transparent;
}

.iconButton.medium{
    width: 35px;
    height: 35px;
}

.iconButton.medium > img {
    height: 37px;
    width: 37px;
}
