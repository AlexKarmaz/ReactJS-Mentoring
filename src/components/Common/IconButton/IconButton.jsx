import React from 'react';
import './IconButton.css';

//PATTERN: Stateless component
const IconButton = ({size='medium', onClick, imgSrc}) => (
    <button className={`iconButton ${size}`} onClick={onClick}>
        <img className='iconButton-image' src={imgSrc} alt='Icon button' />
    </button>
);

export default IconButton;
