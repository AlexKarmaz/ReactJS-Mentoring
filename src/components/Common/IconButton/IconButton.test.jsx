import { render, shallow } from 'enzyme';
import React from 'react';
import IconButton from './IconButton';

describe('IconButton', () => {
    it('should be rendered', () => {
        const onClick = jest.fn();

        const iconButton = render(<IconButton onClick={onClick} />);

        expect(iconButton).toMatchSnapshot();
    });

    it('should receive correct props', () => {
        const onClick = jest.fn();
        const iconSrc =  '/icon.png';

        const iconButton = shallow(<IconButton onClick={onClick} imgSrc={iconSrc} />);
        iconButton.simulate('click');

        expect(iconButton.find('button').hasClass('iconButton medium')).toBeTruthy();
        expect(iconButton.find('img').prop('src')).toEqual(iconSrc);
        expect(onClick).toBeCalled();
    });
});
