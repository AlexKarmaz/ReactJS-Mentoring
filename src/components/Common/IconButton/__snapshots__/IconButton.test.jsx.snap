// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "iconButton medium",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "iconButton-image",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
